File: java_pro/src/main/java/jv/com/tencent/cain/design_pattern/proxy/ProxyPattern.java
package jv.com.tencent.cain.design_pattern.proxy;

/**
 * @author : jiangyu
 * @date : 2019-05-18
 * @desc : 代理设计模式
 */
public class ProxyPattern {
    public static void main(String[] args) {
        SchoolGirl schoolGirl = new SchoolGirl("Lily");
        Proxy giveGiftProxy = new Proxy(schoolGirl);
        giveGiftProxy.giveDolls();
        giveGiftProxy.giveFlowers();
        giveGiftProxy.giveChocolate();
    }

}

File: java_pro/src/main/kotlin/kt/com/tencent/cain/desigin_pattern/proxy/ProxyPattern.kt
package kt.com.tencent.cain.desigin_pattern.proxy

/**
 * @author : jiangyu
 * @date   : 2019-05-18
 * @desc   : 代理设计模式
 */
fun main() {
    val schoolGirl = SchoolGirl("Linda")
    val proxy = Proxy(schoolGirl)
    proxy.giveChocolate()
    proxy.giveDolls()
    proxy.giveFlowers()
}